sklearn/gp_search.py

```diff
@@ -0,0 +1,420 @@
+"""
+The :mod:`sklearn.gp_search` includes utilities to fine-tune the parameters
+of an estimator through a Gaussian Process model.
+"""
+from __future__ import print_function
+
+# Author: Sebastien Dubois <http://bit.ly/SebastienDubois>,
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.stats import norm
+
+from .gaussian_process.gaussian_process import GaussianProcess
+from .cross_validation import check_cv
+from .cross_validation import _fit_and_score
+from .metrics.scorer import check_scoring
+from .base import is_classifier, clone
+
+
+#   UTILS    #
+
+def sample_candidates(n_candidates, param_bounds, param_isInt):
+
+    n_parameters = param_isInt.shape[0]
+    candidates = []
+
+    for k in range(n_parameters):
+        if(param_isInt[k]):
+            k_sample = np.asarray(
+                np.random.rand(n_candidates)
+                * np.float(param_bounds[k][1]-param_bounds[k][0])
+                + param_bounds[k][0],
+                dtype=np.int32)
+        else:
+            k_sample = np.asarray(
+                np.random.rand(n_candidates)
+                * np.float(param_bounds[k][1]-param_bounds[k][0])
+                + param_bounds[k][0])
+        candidates.append(k_sample)
+
+    candidates = np.asarray(candidates)
+    candidates = candidates.T
+
+    return compute_unique(candidates)
+
+
+def compute_ei(predictions, sigma, y_best):
+    ei_array = np.zeros(predictions.shape[0])
+    for i in range(ei_array.shape[0]):
```

Comment:
please try to avoid explicit for-loops if possible. it should be possible to compute ei_array with a single expression using vectorized code